### PR TITLE
[BEAM-330] Port cleanupDaemonThreads fix to archetype module

### PR DESCRIPTION
**Pull request by swegner**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by swegner** [comment]
R: @dhalperi 
CC: @kennknowles

**Comment by lukecwik** [comment]
Don't we also want this in the starter module in addition to the examples module?

File: sdks/java/maven-archetypes/examples/src/main/resources/archetype-resources/pom.xml [hunk 1]
```diff
@@ -88,6 +88,17 @@
         </dependencies>
       </plugin>
     </plugins>
+
+    <pluginManagement>
```
Comment (swegner) [inline sdks/java/maven-archetypes/examples/src/main/resources/archetype-resources/pom.xml]: This is what I expect is the minimal configuration needed from the [parent pom.xml](https://github.com/GoogleCloudPlatform/DataflowJavaSDK/blob/master/pom.xml#L164). I'm not familiar with archetype projects; @dhalperi, do you know how to test that this works?

Comment (kennknowles) [inline sdks/java/maven-archetypes/examples/src/main/resources/archetype-resources/pom.xml]: Looks like there is enough testing that this already failed. So that's good news. To fix the issue that showed up, you'll need another layer of `<plugins></plugins>` between `<pluginManagement>` and `<plugin>`.

**Comment by dhalperi** [comment]
R: @lukecwik -@dhalperi

Luke should review this and make sure it's all set.

**Comment by swegner** [comment]
Ok, this is finally working. @lukecwik PTAL

Verified by locally changing the Beam pom.xml versions from 0.1-incubating-SNAPSHOT to 0.1-incubating, installing, and running the following commands:

mvn archetype:generate -DarchetypeGroupId=org.apache.beam   -DarchetypeArtifactId=beam-sdks-java-maven-archetypes-examples   -DgroupId=com.example -DartifactId=second-dataflow -Dversion="[1.0.0,2.0.0)"   -DinteractiveMode=false

mvn compile exec:java \
  -Dexec.mainClass=com.example.WordCount \
  -Dexec.args="--project=cloud-millwheel \
  --stagingLocation=gs://dataflow-cloud-millwheel/staging/ \
  --output=gs://dataflow-cloud-millwheel/output \
  --runner=BlockingDataflowPipelineRunner"

File: sdks/java/maven-archetypes/examples/src/test/resources/projects/basic/archetype.properties [hunk 2]
```diff
@@ -15,7 +15,7 @@
 #    limitations under the License.
 #
 package=it.pkg
-version=0.1-SNAPSHOT
+version=0.1
```
Comment (dhalperi) [inline sdks/java/maven-archetypes/examples/src/test/resources/projects/basic/archetype.properties]: why this change? I think SNAPSHOT should be there.

But maybe it should be 0.2-SNAPSHOT?

File: sdks/java/maven-archetypes/starter/src/test/resources/projects/basic/archetype.properties [hunk 3]
```diff
@@ -15,7 +15,7 @@
 #    limitations under the License.
 #
 package=it.pkg
-version=0.1-SNAPSHOT
+version=0.1
```
Comment (dhalperi) [inline sdks/java/maven-archetypes/starter/src/test/resources/projects/basic/archetype.properties]: similar comments

File: sdks/java/maven-archetypes/starter/src/test/resources/projects/basic/reference/pom.xml [hunk 4]
```diff
@@ -22,7 +22,7 @@
 
   <groupId>archetype.it</groupId>
   <artifactId>basic</artifactId>
-  <version>0.1-SNAPSHOT</version>
+  <version>0.1</version>
```
Comment (dhalperi) [inline sdks/java/maven-archetypes/starter/src/test/resources/projects/basic/reference/pom.xml]: and more more.

File: sdks/java/maven-archetypes/examples/src/test/resources/projects/basic/archetype.properties (hunk 2 above)
Comment (swegner) [inline sdks/java/maven-archetypes/examples/src/test/resources/projects/basic/archetype.properties]: This is the template pom.xml that will become part of the user's project, and thus should be separate from the versioning of Beam. The versioning for the user project is somewhat arbitrary, but it should represent "the first version for a new project", so 0.1 makes sense to me.

At any rate, we don't need this change coupled to this bug fix. I'll remove it here and file a separate bug.

Comment (swegner) [inline sdks/java/maven-archetypes/examples/src/test/resources/projects/basic/archetype.properties]: [[BEAM-331] Archetype project version shouldn't be coupled to Beam version](https://issues.apache.org/jira/browse/BEAM-331)

File: examples/java/pom.xml [hunk 5]
```diff
@@ -22,7 +22,7 @@
   <parent>
     <groupId>org.apache.beam</groupId>
     <artifactId>beam-examples-parent</artifactId>
-    <version>0.2.0-incubating-SNAPSHOT</version>
```
Comment (lukecwik) [inline examples/java/pom.xml]: I think you want to revert the pom files which you edited to validate the change.

**Comment by swegner** [comment]
@lukecwik PTAL; I believe this should be ready to go in once Jenkin goes green

File: sdks/java/maven-archetypes/examples/src/test/resources/projects/basic/archetype.properties (hunk 2 above)
Comment (swegner) [inline sdks/java/maven-archetypes/examples/src/test/resources/projects/basic/archetype.properties]: Make that [[BEAM-339]](https://issues.apache.org/jira/browse/BEAM-339).

**Comment by lukecwik** [comment]
LGTM, will merge after checks go green

**Comment by swegner** [comment]
Backport: https://github.com/GoogleCloudPlatform/DataflowJavaSDK/pull/302
